### DISCUSSION TITLE
Remove Bookmarked Lines/ Remove Non-Bookmarked Lines

### DISCUSSION THREAD

**Discussion by Kajmoswega** [alefragnani/vscode-bookmarks#711] [Ideas]
Hi, would be possible make new features? In notepad++ I used features for productivity as remove bookmarked lines or remove non-bookmarked lines in .csv documents. Also feature as "find in project and marked all lines" would be great :) If you have questions about details don't be afraid to ask me.

**Comment by alefragnani** [comment]
Hi @Kajmoswega ,

Of course it is possible to add new features, but it needs to follow the extension's main scope. I try to keep the extension dealing with bookmarks only, so sometimes you would need to _combine_ with other commands or extensions.

* **remove bookmarked lines**: I'm not sure you are aware of the `Bookmarks: Select Lines` command. Use it prior to the builtin `Delete Line`, and you will have the same result
* **remove non-bookmarked lines**: You could use the same `Bookmarks: Select Lines` command, combined with some _revert selection_ extension, or even a _delete other lines_ command/extension. I wonder if wouldn't exist one already
* **find in project and marked all lines**: This one is tricky. The extension supports multi-cursor, so when you use the `Bookmarks: Toggle` command in a multi-cursor editing, all lines in the current file that contains a cursor will be bookmarked. The _find in project_, on the other hand, I wound't support. Not only because it goes outside the scope of the extension, but also because the current Search API provided by VS Code does not give the extensions the info it needs. Instead, it only provides a way to extensions to create their own search, which is something I don't want to add to the extension.

Hope this helps
